code/_onclick/click.dm

```diff
@@ -331,16 +331,19 @@
 	else
 		..()
 
-/atom/proc/AltClick(var/mob/user)
+/atom/proc/AltClick(mob/user)
+	SEND_SIGNAL(src, COMSIG_CLICK_ALT, user)
 	var/turf/T = get_turf(src)
-	if(T)
-		if(user.TurfAdjacent(T))
-			user.listed_turf = T
-			user.client.statpanel = T.name
-			// If we had a method to force a `Stat` update, it would go here
-		else
-			user.listed_turf = null
-	return
+	if(T && (isturf(loc) || isturf(src)) && user.TurfAdjacent(T))
```

Comment:
I don't see what the `(isturf(loc) || isturf(src))` checks are trying to accomplish here, since one of these should always be true theoretically.

A) You alt-click on some atom thats sitting on a turf, `isturf(loc)` is true: **success**
B) You alt-click on a turf directly, `isturf(src)` is true: **success**
C) You alt-click on an atom which is inside something, like a locker, `(isturf(loc) || isturf(src))` are both false: **failure**

So scenario C is the only time you can fail, but it's impossible for this to happen, due to you not being able to see the atom to alt-click on, since it's inside another atom.

Reply:
>So scenario C is the only time you can fail, but it's impossible for this to happen, due to you not being able to see the atom to alt-click on, since it's inside another atom.

Objects in your inventory or your backpack are neither a turf nor is their `loc` a turf.

Reply:
Yeah you're right, and you know what.. I had no idea a `get_turf(src)` of an item in a backpack ultimately returned the turf it's on, I thought it would be null. There must be something going on behind the scenes I wasn't aware of.